test/org/sagebionetworks/bridge/play/controllers/ConsentControllerMockedTest.java

```diff
@@ -163,8 +167,14 @@ public void canWithdrawConsent() throws Exception {
         JsonNode node = BridgeObjectMapper.get().readTree(response);
         assertEquals("User has been withdrawn from the study.", node.get("message").asText());
         
+        // Should call the service and withdraw
         verify(consentService).withdrawConsent(any(Study.class), any(User.class), captor.capture(), any(Long.class));
         assertEquals("Because, reasons.", captor.getValue().getReason());
+        
+        // Should update the user's session to be non-consented
+        ArgumentCaptor<UserSession> sessionCaptor = ArgumentCaptor.forClass(UserSession.class);
+        verify(cacheProvider).setUserSession(sessionCaptor.capture());
+        assertFalse(sessionCaptor.getValue().getUser().doesConsent());
```

Comment:
Moving this to the service test where the service is not mocked but the user is, and verifying the correct methods are called on the user object.